src/components/navigation/document-tab-panel.tsx

```diff
@@ -0,0 +1,154 @@
+      >
+        <TabList className={`tab-list ${leftTabSpec?.tab}`} onClick={onTabClick}>
+          {leftTabSpec?.sections.map((section) => {
+            const sectionTitle = this.getSectionTitle(section, this.stores);
+            return (
+              <Tab
+                className={`doc-tab ${leftTabSpec?.tab} ${section.type}`}
+                key={`section-${section.type}`}
+                data-test={section.dataTestHeader}
+              >
+                {sectionTitle}
+              </Tab>
+            );
+          })}
+        </TabList>
+        {leftTabSpec?.sections.map((section, index) => {
+          const _handleDocumentStarClick = section.showStarsForUser(user)
+                ? this.handleDocumentStarClick
+                : undefined;
+          const _handleDocumentDeleteClick = section.showDeleteForUser(user)
+                ? this.handleDocumentDeleteClick
+                : undefined;
+          return (
+            <TabPanel key={`section-${section.type}`}>
+              { document && (index === tabIndex)
+                ? <EditableDocumentContent
+                    mode={"1-up"}
+                    isPrimary={false}
+                    document={document}
+                    readOnly={true}
+                  />
+                : <TabPanelDocumentsSection
+                    key={section.type}
+                    tab={leftTabSpec!.tab}
+                    section={section}
+                    stores={this.stores}
+                    scale={kNavItemScale}
+                    onNewDocumentClick={this.handleNewDocumentClick}
+                    onDocumentClick={onDocumentClick || this.handleDocumentClick}
+                    onDocumentDragStart={this.handleDocumentDragStart}
+                    onDocumentStarClick={_handleDocumentStarClick}
+                    onDocumentDeleteClick={_handleDocumentDeleteClick}
+                  /> }
+            </TabPanel>
+          );
+        })}
+      </Tabs>
+    );
+  }
+
+  private getSectionTitle = (section: LeftTabSectionModelType, stores: IStores) => {
+    if (section.title === "%abbrevInvestigation%") {
+      const { unit, investigation } = stores;
+      const { abbrevTitle } = unit;
+      const prefix = abbrevTitle ? `${abbrevTitle}: ` : "";
+      return `${prefix}Investigation ${investigation.ordinal}`;
+    }
+    return section.title;
+  }
+
+  private handleNewDocumentClick = async (section: LeftTabSectionModelType) => {
+    const { appConfig: { defaultDocumentContent }, db, ui } = this.stores;
+    const { problemWorkspace } = ui;
+    const newDocument = section.type === ELeftTabSectionType.kPersonalDocuments
+                          ? await db.createPersonalDocument({ content: defaultDocumentContent })
+                          : await db.createLearningLogDocument();
+    if (newDocument) {
+      problemWorkspace.setAvailableDocument(newDocument);
+      ui.restoreDefaultNavExpansion();
+    }
+  }
```

Comment:
As mentioned above, ultimately I think we should remove all of these default handlers. Also, it seems like this would do the wrong thing when creating a new document from the Problem Document section. I would think that this would be safer:
```typescript
    const newDocument = section.type === ELeftTabSectionType.kLearningLogDocuments
                          ? await db.createLearningLogDocument()
                          : await db.createPersonalDocument({ content: defaultDocumentContent });
```

Reply:
Fixed.